controllers/storagecluster/cephcluster.go

```diff
@@ -491,6 +491,18 @@ func newCephCluster(sc *ocsv1.StorageCluster, cephImage string, kmsConfigMap *co
 		},
 	}
 
+	if sc.Spec.ManagedResources.CephCluster.ContinueUpgradeAfterChecksEvenIfNotHealthy {
```

Comment:
It should be same result even if you don't check these conditions, and always set the bool. 
```suggestion
```

Reply:
So, in that case we will have to define default values in the CR, and that will lead to those fields present in the storagecluster CR by default and visible to everyone, which we don't want (as per the last discussion).
That's the reason I added this condition to only update, if present in SC CR

Reply:
Perhaps that only applies to the `bool?` type? A `bool` already has a default value of false, so I don't see how this will change customer visibility.

Reply:
Right, agreed.
In the epic, description for `ContinueUpgradeAfterChecksEvenIfNotHealthy` the default value mentioned is `true`, but I think that's not the case of go by the usage of the property? If so, then we can remove the conditionals but, if that's not the case then we would have to define `default: true` in the spec and that would make the property visible in the CR after creation. So, just need a clarity on that.

Reply:
Actually, `ContinueUpgradeAfterChecksEvenIfNotHealthy` is a special case since its default value is defined up on line 475. So for this setting we would need to define the StorageCluster as a `bool?` and only override the default if explicitly set. So we just as well take the same approach and use `bool?` for the other two settings as well.

Reply:
I have defined all of them as bool types only, and override the default if set in the StorageCluster CR.
So, are you suggesting that I should just do something like this:
```
         cephCluster := &rookCephv1.CephCluster {
                      .....
                      .....
                      WaitTimeoutForHealthyOSDInMinutes: getWaitTimeoutForHealthOSD(sc),
                      SkipUpgradeChecks: sc.Spec.ManagedResources.CephCluster.SkipUpgradeChecks
                      UpgradeOSDRequiresHealthyPGs: sc.Spec.ManagedResources.CephCluster.UpgradeOSDRequiresHealthyPGs
		},
	}

	if sc.Spec.ManagedResources.CephCluster.ContinueUpgradeAfterChecksEvenIfNotHealthy {
		cephCluster.Spec.ContinueUpgradeAfterChecksEvenIfNotHealthy = sc.Spec.ManagedResources.CephCluster.ContinueUpgradeAfterChecksEvenIfNotHealthy
	}
```

Reply:
Clarification... I meant to define a `bool*` in the StorageCluster CRD definition. The problem for a `bool` is that we can't tell the difference between the user setting it to `false`, or if it is not specified at all. With a `bool*`, the check would be something like:
```
if sc.Spec.ManagedResources.CephCluster.ContinueUpgradeAfterChecksEvenIfNotHealthy != nil {
		cephCluster.Spec.ContinueUpgradeAfterChecksEvenIfNotHealthy = *sc.Spec.ManagedResources.CephCluster.ContinueUpgradeAfterChecksEvenIfNotHealthy
	}
```

Looks good to set the other two values above since they already have `false` for a default.

Reply:
Made the changes.